api/src/opentrons/protocol_api/instrument_context.py

```diff
@@ -432,50 +447,48 @@ def blow_out(
                               :py:meth:`dispense`)
         :returns: This instance
         """
-
         well: Optional[labware.Well]
-        # TODO(jbl 2022-11-10) refactor this boolean out and make location optional when PE blow-out in place exists
-        move_to_well = True
-        last_location = self._protocol_core.get_last_location()
+        move_to_location: types.Location
+        blow_out_in_place: bool = False
 
-        if isinstance(location, labware.Well):
-            if location.parent.is_tiprack:
-                _log.warning(
-                    "Blow_out being performed on a tiprack. "
-                    "Please re-check your code"
-                )
-            checked_loc = location.top()
-            well = location
-        elif isinstance(location, types.Location):
-            checked_loc = location
-            _, well = location.labware.get_parent_labware_and_well()
-        elif location is not None:
-            raise TypeError(
-                "location should be a Well or Location, but it is {}".format(location)
+        last_location = self._get_last_location_by_api_version()
+        try:
+            target = validation.validate_location(
+                location=location, last_location=last_location
             )
-        elif last_location:
-            checked_loc = last_location
-            _, well = checked_loc.labware.get_parent_labware_and_well()
-            # if no explicit location given but location cache exists,
-            # pipette blows out immediately at
-            # current location, no movement is needed
-            move_to_well = False
-        else:
+        except validation.NoLocationError:
             raise RuntimeError(
                 "If blow out is called without an explicit location, another"
                 " method that moves to a location (such as move_to or "
                 "dispense) must previously have been called so the robot "
                 "knows where it is."
             )
+        except validation.LocationTypeError:
+            raise TypeError(
+                "location should be a Well or Location, but it is {}".format(location)
+            )
```

Comment:
```suggestion
        except validation.LocationTypeError as e:
            raise TypeError(
                f"location should be a Well or Location, but it is {location}"
            ) from e
```

This particular error message is exactly the same between aspirate, dispense, and blow out. I'd be open to moving the message into the original `raise LocationTypeError` and then letting it propagate without an `except`